tests/json_front_tests/test_json_class.py

```diff
@@ -0,0 +1,170 @@
+
+
+class TestJsonClass(unittest.TestCase):
+    """Test the JSON frontend for the Causal Testing Framework (CTF)
+
+    The JSON frontend is an alternative interface for the CTF where tests are specified in JSON format and ingested
+    with the frontend. Tests involve testing that this correctly interfaces with the framework with some dummy data
+    """
+
+    def setUp(self) -> None:
+        temp_dir_path = Path(create_temp_dir_if_non_existent())
+        json_file_name = "tests.json"
+        dag_file_name = "dag.dot"
+        data_file_name = "data.csv"
+        self.json_path = temp_dir_path / json_file_name
+        self.dag_path = temp_dir_path / dag_file_name
+        self.data_path = temp_dir_path / data_file_name
+        setup_files(self.json_path, self.data_path, self.dag_path)
+        self.json_class = JsonUtility("logs.log")
+        self.example_distribution = scipy.stats.uniform(1, 10)
+        self.input_dict_list = [{"name": "test_input", "type": float, "distribution": self.example_distribution}]
+        self.output_dict_list = [{"name": "test_output", "type": float}]
+        self.meta_dict_list = [{"name": "test_meta", "type": float, "populate": populate_example}]
+        self.json_class.set_variables(self.input_dict_list, self.output_dict_list, None)
+        self.json_class.set_path(self.json_path, self.dag_path, self.data_path)
+
+    def test_setting_paths(self):
+        self.assertEqual(self.json_class.json_path, Path(self.json_path))
+        self.assertEqual(self.json_class.dag_path, Path(self.dag_path))
+        self.assertEqual(self.json_class.data_path, Path(self.data_path))
+
+    def test_set_inputs(self):
+        ctf_input = [Input("test_input", float, self.example_distribution)]
+        self.assertEqual(ctf_input[0].name, self.json_class.inputs[0].name)
+        self.assertEqual(ctf_input[0].datatype, self.json_class.inputs[0].datatype)
+        self.assertEqual(ctf_input[0].distribution, self.json_class.inputs[0].distribution)
+
+    def test_set_outputs(self):
+        ctf_output = [Output("test_output", float)]
+        self.assertEqual(ctf_output[0].name, self.json_class.outputs[0].name)
+        self.assertEqual(ctf_output[0].datatype, self.json_class.outputs[0].datatype)
+
+    def test_set_metas(self):
+        self.json_class.set_variables(self.input_dict_list, self.output_dict_list, self.meta_dict_list)
+        ctf_meta = [Meta("test_meta", float, populate_example)]
+        self.assertEqual(ctf_meta[0].name, self.json_class.metas[0].name)
+        self.assertEqual(ctf_meta[0].datatype, self.json_class.metas[0].datatype)
+
+    def test_argparse(self):
+        args = self.json_class.get_args(["--data_path=data.csv", "--dag_path=dag.dot", "--json_path=tests.json"])
+        self.assertTrue(args.data_path)
+        self.assertTrue(args.dag_path)
+        self.assertTrue(args.json_path)
+
+    def test_setup_modelling_scenario(self):
+        self.json_class.setup()
```

Comment:
It would be cleaner code wise to place it in setUp. It's quite computationally expensive to call though and on my laptop roughly doubles the test execute time to call with each test from ~1.6 s to ~3s . Not sure what is better in this tradeoff?

Reply:
Is `setUp` only called once, or for each test?

Reply:
For each test currently

Reply:
Would things like the path file names be better off as global variables? Rather than being redefined for each test

Reply:
I think best practise is to use `setUp`, but I mostly use `pytest` where this would probably be a fixture.